### PR TITLE
Add workflow for incline service

### PR DESCRIPTION
**Pull request by sujata-m**
**Incline Servic:** 

- The service would take a `dataset_id`
- It will fetch the `latest_dataset_url` from DB for that particular `dataset_id`
    ```
    const dataset = await this.tdeiCoreServiceInstance.getDatasetDetailsById(backendRequest.parameters.dataset_id);
    ```
- Service will trigger the workflow:
  - Sends the message in `osw-incline-request` topic with payload
      ```json
      "jobId": "<%=JSON.stringify(workflow_input.job_id)%>",
      "user_id": "<%=workflow_input.user_id%>",
      "dataset_url": "<%=workflow_input.dataset_url%>"
      ```
  - Once it receives the output:
     ```json
      "success": "<%=data.success%>",
      "message": "<%=data.message%>",
      "file_upload_path": "<%=data.file_upload_path%>"
      ```
  - Update the DB with `latest_dataset_url` got from output
  - Does the OSW to OSM formatting
  - Update the `latest_osm_url` with latest inclination tags
  - Update the `dataset_osm_download_url` 
  - Update the `dataset_download_url`
  - Finally Update the job status to `Completed`

### CODE REVIEW

File: src/controller/osw-controller.ts [hunk 1]
```diff
@@ -144,6 +144,7 @@ class OSWController implements IController {
         // Route for quality metric request
         this.router.post(`${this.path}/quality-metric/:tdei_dataset_id`, qualityUpload.single('file'),authenticate, this.createQualityOnDemandRequest);
         this.router.post(`${this.path}/quality-metric/tag/:tdei_dataset_id`, tagQuality.single('file'), authenticate, this.tagQualityMetric);
+        this.router.post(`${this.path}/dataset-add-inclination/`, authenticate, this.createInclineRequest);
```
Comment (MashB) [inline src/controller/osw-controller.ts:147]: rename dataset-add-inclination => dataset-inclination . POST suggests it is add.

Comment (MashB) [reply src/controller/osw-controller.ts:147]: Remove trailing / from url

File: src/controller/osw-controller.ts [hunk 2]
```diff
@@ -565,6 +566,61 @@ class OSWController implements IController {
...
+            }
+
+            if (requestService.dataset_id == undefined) {
+                return next(new InputException('required input is empty', response));
+            }
+
+            let apiKey = request.headers['x-api-key'];
+            //Reject authorization for API key users
+            if (apiKey && apiKey !== '') {
+                return next(new UnAuthenticated());
```
Comment (MashB) [inline src/controller/osw-controller.ts:593]: Return ForbiddenAccess() ,

File: src/controller/osw-controller.ts (hunk 1 above)
Comment (MashB) [reply src/controller/osw-controller.ts:147]: If dataset_id is mandatory and inclination is for dataset_id , then accept the input as part of path variable

File: src/tdei-orchestrator-config_v2.json [hunk 3]
```diff
@@ -2065,6 +2065,227 @@
...
+                    },
+                    "function": "update_table"
+                },
+                {
+                    "name": "get_incline_blob_folder_path_wo_ext",
+                    "task_reference_name": "get_incline_blob_folder_path_wo_ext",
+                    "description": "Getting the blob folder path for zip upload",
+                    "type": "Utility",
+                    "input_params": {
+                        "blobUrl": "<%=workflow_input.dataset_url%>",
```
Comment (MashB) [inline src/tdei-orchestrator-config_v2.json:2162]: should'nt be tasks.osw_dataset_incline_tag.output.file_upload_path ?

**Review by MashB — CHANGES_REQUESTED** [review]

File: src/controller/osw-controller.ts [hunk 4]
```diff
@@ -565,6 +566,61 @@ class OSWController implements IController {
...
+            var authorized = await Utility.authorizeRoles(request.body.user_id, osw.tdei_project_group_id, ["tdei_admin", "poc", "osw_data_generator"]);
+            if (!authorized) {
+                return next(new ForbiddenAccess());
+            }
+
+            let backendRequest: InclinationServiceRequest = {
+                user_id: request.body.user_id,
+                service: "add_inclination",
+                parameters: {
+                    dataset_id: requestService.dataset_id
```
Comment (MashB) [inline src/controller/osw-controller.ts:607]: update only dataset id

File: src/service/osw-service.ts [hunk 5]
```diff
@@ -1048,6 +1048,55 @@ class OswService implements IOswService {
...
+
+            let job = CreateJobDTO.from({
+                data_type: TDEIDataType.osw,
+                job_type: JobType["Dataset-Incline-Tag"],
+                status: JobStatus["IN-PROGRESS"],
+                message: 'Job started',
+                request_input: {
+                    service: backendRequest.service,
+                    user_id: backendRequest.user_id,
+                    parameters: backendRequest.parameters
```
Comment (MashB) [inline src/service/osw-service.ts:1074]: update dataset id only

File: src/service/osw-service.ts [hunk 6]
```diff
@@ -1048,6 +1048,55 @@ class OswService implements IOswService {
...
+            });
+
+            const job_id = await this.jobServiceInstance.createJob(job);
+            
+            let workflow_start = WorkflowName.osw_dataset_incline_tag;
+            let workflow_input = {
+                job_id: job_id.toString(),
+                service: backendRequest.service,
+                user_id: backendRequest.user_id,
+                parameters: backendRequest.parameters,
```
Comment (MashB) [inline src/service/osw-service.ts:1087]: remove parameters its not used

File: src/tdei-orchestrator-config_v2.json [hunk 7]
```diff
@@ -2065,6 +2065,227 @@
...
                 }
             ]
+        },
+        {
+            "name": "osw_dataset_incline_tag",
+            "description": "Update the dataset with incline tags",
+            "workflow_input": {
+                "service": "<%=service %>",
+                "user_id": "<%=user_id%>",
+                "parameters": "<%= JSON.stringify(parameters) %>",
```
Comment (MashB) [inline src/tdei-orchestrator-config_v2.json:2075]: parameters not used remove

File: src/tdei-orchestrator-config_v2.json [hunk 8]
```diff
@@ -2065,6 +2065,227 @@
...
+                    "function": "get_blob_folder_path_wo_ext"
+                },
+                {
+                    "name": "osm_compression_incline_tag",
+                    "task_reference_name": "osm_compression_incline_tag",
+                    "description": "Compressing the osm file",
+                    "type": "Event",
+                    "topic": "dataset-zip-request",
+                    "input_params": {
+                        "input_urls": "<%= JSON.stringify([decodeURIComponent(tasks.osw_incline_tag_osm_formatting.output.formatted_url), workflow_input.metadata_url, workflow_input.changeset_url]) %>",
```
Comment (MashB) [inline src/tdei-orchestrator-config_v2.json:2179]: changeset_url , metaurl missing in input please add

**Review by MashB — CHANGES_REQUESTED** [review]

File: src/service/interface/osw-service-interface.ts [hunk 9]
```diff
@@ -142,4 +142,14 @@ export interface IOswService {
...
+
+
+    /**
+     * Calculates inclination for a given TDEI dataset.
+     * 
+     * @param backendRequest The backend request to process.
+     * @returns A Promise that resolves to a string representing the job ID.
+     * @throws Throws an error if an error occurs during processing.
+     */
+    calculateInclanation(backendRequest: InclinationServiceRequest): Promise<string>;
```
Comment (susrisha) [inline src/service/interface/osw-service-interface.ts:154]: spell check. the method name should be `calculateInclination`

File: src/service/osw-service.ts [hunk 10]
```diff
@@ -1048,6 +1048,54 @@ class OswService implements IOswService {
...
         }
     }
+
+    /**
+    * Processes a inclination request by uploading a file, creating a job, triggering a workflow, and returning the job ID.
+    * @param backendRequest The backend request to process.
+    * @returns A Promise that resolves to a string representing the job ID.
+    * @throws Throws an error if an error occurs during processing.
+    */
+    async calculateInclanation(backendRequest: InclinationServiceRequest): Promise<string> {
```
Comment (susrisha) [inline src/service/osw-service.ts:1058]: please rename the method. there is spelling mistake

**Review by susrisha — CHANGES_REQUESTED** [review]
Spelling checks and also need to rebase

**Pull request merged**